script/cibuild-setup-py: install all dev requirements for testing

z-bsod opened the pull request:
install all requirements stored in requirements-dev instead of only a short hardcoded list for the test of the test after build

Commit:
script/cibuild-setup-py: install all dev requirements for testing

install all requirements stored in requirements-dev instead of only a
short hardcoded list

ross commented:
> install all requirements stored in requirements-dev instead of only a short hardcoded list for the test of the test after build

It intentionally doesn't install requirements-dev.txt before running tests to prevent things in that file from being accidental production/runtime dependencies. Those requirements are test-time and rather than development related. 

You can see the distinction in setup.py, https://github.com/octodns/octodns-template/blob/7011a39d5f2649b624b446232692dd79b56edba6/setup.py#L23-L28

and 

https://github.com/octodns/octodns-template/blob/7011a39d5f2649b624b446232692dd79b56edba6/setup.py#L34-L49

Iirc i looked into installing the test specific requirements with pip in the past with something like

```console
pip install dist/*$VERSION*.whl[test]
```

but ran into problems/bugs that prevented it from functioning correctly. Even then it would allow things in the test section to be accidental "production" requirements so there's a small(er) risk to that.

ross commented:
Oh, forgot to mention, setup.py also explicitly doesn't use requirements.txt and sources everything from setup.py since that's what it's specifically testing. requirements*.txt is used for all the other testing.

z-bsod committed:
cibuild-setup-py

install test dependencies from setup.py / built wheel file before running pytest

z-bsod commented:
i was aware of the difference between the requirements for test and dev in setup.py.

If you list the same test requirements in the ci script as in the `requirements_tests` in the `setup.py` file there should be no difference regarding accidential requirements to test dependencies but i don't see any way of resolving this as you can't run the tests without the test requirements.

i just tried running the tests with `pip install "${wheelfile}[test]"` and it seems to work correctly the only issue is that pip doesn't support file globbing when executed this way.

I updated the script in my MR and tried the modified version in the repo of the autodns-provider and it does seem to work correctly there ([example run](https://github.com/telekom-mms/octodns-autodns/actions/runs/12654115473/job/35261221954?pr=6))

ross reviewed (approved):
> i just tried running the tests with `pip install "${wheelfile}[test]"` and it seems to work correctly the only issue is that pip doesn't support file globbing when executed this way.

:cool: It was long enough ago that I don't remember what problems(s) I hit with it. Maybe this was it or it was another since addressed shortcoming with pip.

> I updated the script in my MR and tried the modified version in the repo of the autodns-provider and it does seem to work correctly there ([example run](https://github.com/telekom-mms/octodns-autodns/actions/runs/12654115473/job/35261221954?pr=6))

👍
